core/src/main/java/kafka/server/share/DelayedShareFetch.java

```diff
@@ -204,7 +217,29 @@ Map<TopicIdPartition, FetchRequest.PartitionData> acquirablePartitions() {
         return topicPartitionData;
     }
 
-    private void releasePartitionLocks(String groupId, Set<TopicIdPartition> topicIdPartitions) {
+    boolean isMinBytesCriteriaSatisfied(Map<TopicIdPartition, FetchRequest.PartitionData> topicPartitionData) {
+        AtomicLong accumulatedSize = new AtomicLong(0);
+        topicPartitionData.forEach((topicIdPartition, partitionData) -> {
+            Partition partition = replicaManager.getPartitionOrException(topicIdPartition.topicPartition());
```

Comment:
How the exception will be handled with this method? Do we have a test case when exception is thrown by this replica manager call?

Reply:
I've added the handling and test case for this in my latest commit